wsit/ws-sx/wssx-impl/src/main/java/com/sun/xml/wss/impl/misc/SecurityUtil.java

```diff
@@ -36,6 +37,7 @@
 import com.sun.xml.wss.XWSSecurityException;
 import com.sun.xml.wss.impl.MessageConstants;
 import com.sun.xml.wss.logging.LogDomainConstants;
+import com.sun.xml.wss.logging.impl.crypto.LogStringsMessages;
```

Comment:
the rule is simple:

if a class is in package `com.sun.xml.wss.impl.misc`, keys come from the resource in `com.sun.xml.wss.logging.impl.misc`; if a class is in package `com.sun.xml.wss.impl.crypto`, keys come from the resource in `com.sun.xml.wss.logging.impl.crypto` - unless some specific message is shared in multiple places

Reply:
I would start thinking about moving this class somewhere else - crypto? security? own package? Depends on class and package dependencies, I did not investigate too much, but "misc" is not the right package for this specialized utility class. Maybe it would make even sense to first remove methods if they are used just from one concrete place. That would help to get rid of part of dependencies and have a clear view on what remained.

Reply:
> Maybe it would make even sense to first remove methods if they are used just from one concrete place.

that would be good start, I don't mind if the same code is copy&pasted on 2 places (3+ is just too much...) should it help